views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/player.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
+ */
+/**
+ * This is an audio player for the sole purpose of playing back the recorded media.
+ * It is actually a very simple wrapper around an HTMLAudioElement
+ *
+ * @author Christophe Noël <christophe@taotesting.com>
+ */
+define([
+    'OAT/util/event'
+], function(event) {
+    'use strict';
+
+    /**
+     * @property {String} CREATED   - player instance created, but no media loaded
+     * @property {String} IDLE      - media loaded and playable
+     * @property {String} PLAYING   - media is playing
+     */
+    var playerStates = {
+        CREATED:    'created',
+        IDLE:       'idle',
+        PLAYING:    'playing'
+    };
+
+    /**
+     * @returns {Object} - The player
+     */
+    return function playerFactory() {
+        var audioEl,
+            player,
+            state = playerStates.CREATED;
+
+        player = {
+            /**
+             * Set player state
+             * @param {String} newState
+             * @private
+             */
+            _setState: function _setState(newState) {
```

Comment:
If this is actually a private method, it could be better to move it as a closure, and use the reference to `player` instead of `this`.